rest_framework/permissions.py

```diff
@@ -103,6 +103,12 @@ def get_required_permissions(self, method, model_cls):
         Given a model and an HTTP method, return the list of permission
         codes that the user is required to have.
         """
+        while model_cls._meta.proxy:
+            if model_cls.__name__ == model_cls._meta.concrete_model.__name__:
```

Comment:
Is there a more robust way to get the underlying concrete model, rather than inspecting the names. This looks a bit odd.